src/main/java/reposense/util/FileUtil.java

```diff
@@ -57,44 +63,55 @@ public static void deleteDirectory(String root) throws IOException {
         }
     }
 
-    public static void unzip(ZipInputStream zipInput, String destinationFolder) {
-        Path directory = Paths.get(destinationFolder);
-
-        // buffer for read and write data to file
-        byte[] buffer = new byte[2048];
-
-        try {
-            Files.createDirectories(directory);
-            ZipEntry entry = zipInput.getNextEntry();
+    /**
+     * Zips all the JSON files contained in the {@code sourcePath} directory.
+     * Creates the zip folder in the {@code outputPath}
+     */
+    public static void zipJson(String outputPath, String sourcePath) {
+        List<Path> allJsonFiles = getFilePath(outputPath + File.separator + sourcePath, ".json");
+
+        int length;
+        try (FileOutputStream fos = new FileOutputStream(outputPath + File.separator + Constants.JSON_ZIP_FILE);
+             ZipOutputStream zos = new ZipOutputStream(fos)) {
+            for (Path jsonFile : allJsonFiles) {
+                try (InputStream is = Files.newInputStream(jsonFile)) {
+                    zos.putNextEntry(new ZipEntry(getChild(jsonFile.toString(), sourcePath + File.separator)));
+                    while ((length = is.read(buffer.array())) > 0) {
+                        zos.write(buffer.array(), 0, length);
+                    }
+                }
+            }
+        } catch (IOException ioe) {
+            logger.log(Level.SEVERE, ioe.getMessage(), ioe);
+        }
+    }
 
-            while (entry != null) {
-                String entryName = entry.getName();
-                Path path = Paths.get(destinationFolder, entryName);
+    /**
+     * Unzips the contents of the {@code zis} and stores in the {@code destinationFolder}
+     */
+    public static void unzip(ZipInputStream zis, String destinationFolder) {
+        ZipEntry entry;
+        try (ZipInputStream zi = zis) {
```

Comment:
As mentioned earlier, can we make `unzip` take in an `InputStream` instead, and create the new `ZipInputStream` here instead?